#### Question 1: Given some sample data, write a program to answer the following: click here to access the required data set
 > On Shopify, we have exactly 100 sneaker shops, and each of these shops sells only one model of shoe. We want to do some analysis of the average order value (AOV). When we look at orders data over a 30 day window, we naively calculate an AOV of $3145.13. Given that we know these shops are selling sneakers, a relatively affordable item, something seems wrong with our analysis.


In [1]:
pip install pandas

Note: you may need to restart the kernel to use updated packages.


In [20]:
# Load essential library
import pandas as pd
import numpy as np

#### Explore the Data

In [4]:
data = pd.read_csv("datascience.csv")
data.sample(10) #shows us random 10 lines of the dataframe to get a general sense of the data

order_id  shop_id  user_id  order_amount  total_items payment_method  \
1480      1481       66      853           161            1           cash   
431        432       98      841           266            2    credit_card   
2584      2585       89      700           392            2           cash   
1433      1434       15      966           306            2    credit_card   
2858      2859       25      836           390            3    credit_card   
2734      2735       70      950           519            3    credit_card   
930        931       69      912           131            1           cash   
4893      4894       36      917           260            2          debit   
3697      3698       42      839           352            1          debit   
650        651       49      857           129            1          debit   

               created_at  
1480  2017-03-10 17:49:03  
431   2017-03-12 18:26:39  
2584  2017-03-21 12:18:52  
1433  2017-03-27 20:46:51  
2858   2017-03-01 8:21:31  
2734  2017-03-01 16:05:59  
930    2017-03-09 1:39:33  
4893   2017-03-30 0:42:12  
3697   2017-03-12 2:45:09  
650    2017-03-10 4:04:46

In [5]:
print(data.info()) #shows us the data type of the 7 columns

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5000 entries, 0 to 4999
Data columns (total 7 columns):
 #   Column          Non-Null Count  Dtype 
---  ------          --------------  ----- 
 0   order_id        5000 non-null   int64 
 1   shop_id         5000 non-null   int64 
 2   user_id         5000 non-null   int64 
 3   order_amount    5000 non-null   int64 
 4   total_items     5000 non-null   int64 
 5   payment_method  5000 non-null   object
 6   created_at      5000 non-null   object
dtypes: int64(5), object(2)
memory usage: 273.6+ KB
None


In [7]:
data.describe() # shows us the statistical summary of the DataFrame

order_id      shop_id      user_id   order_amount  total_items
count  5000.000000  5000.000000  5000.000000    5000.000000   5000.00000
mean   2500.500000    50.078800   849.092400    3145.128000      8.78720
std    1443.520003    29.006118    87.798982   41282.539349    116.32032
min       1.000000     1.000000   607.000000      90.000000      1.00000
25%    1250.750000    24.000000   775.000000     163.000000      1.00000
50%    2500.500000    50.000000   849.000000     284.000000      2.00000
75%    3750.250000    75.000000   925.000000     390.000000      3.00000
max    5000.000000   100.000000   999.000000  704000.000000   2000.00000

#### a. Think about what could be going wrong with our calculation. Think about a better way to evaluate this data.
> From the statistical summary, the mean of the column 'order_amount' is equivalent to the AOV that we orignially calculated, in this case is $3145.123. 


> To get a better intution of what goes wrong with our caclulation, how we orignally got the wrong caclulation is listed as below:

In [17]:
wrong_aov = data['order_amount'].sum() / data['total_items'].count()
np.round(wrong_aov, decimals = 2)

3145.13

#### b. What metric would you report for this dataset?
> Instead of using the built-in function count( ) to count the number of entries, we should use the built-in fuction sum( ) to caclulate the number of orders. 

In [19]:
correct_aov = data['order_amount'].sum()/data['total_items'].sum()
np.round(correct_aov, decimals = 2) #keep the decimal place to 2 to align with our intended calculation

357.92

#### c. What is its value?
> The correct average order value (AOV) is $357.92